protocol/api/record_batch.go

```diff
@@ -233,11 +219,13 @@ type Record struct {
 	Headers        []RecordHeader
 }
 
-func (r Record) Encode(pe *encoder.Encoder) {
-	pe.PutVarint(int64(r.getEncodedLength())) // Length placeholder
+func (r Record) EncodeFull(pe *encoder.Encoder) {
```

Comment:
Why does this get an "Encode" and "EncodeFull" instead of just Encode? If it's about satisfying the interface so you can use `GetEncodedLength`, we're doing this wrong

Reply:
Don't seem like a good enough reason to pollute a public interface. Haven't thought about how this would work with the encoder thing we're using but if we were using a simple `Encode() []byte` interface for example, I'd just use a private method `encodeContents()` and then call something like `encodeLength() + encodeContents()` within `Encode()`. That way the public interface is simple and the same as all other "encodable" entities we have.